### PR TITLE
Fix ProduceOnlyReferenceAssembly

### LINKED ISSUE #16910: ProduceOnlyReferenceAssembly=true fails with `Do not use refout when using refonly`
The following project fails to compile:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <ProduceOnlyReferenceAssembly>true</ProduceOnlyReferenceAssembly>
  </PropertyGroup>

</Project>
```
The above project will fail to compile with error
> CSC : error CS8308: Do not use refout when using refonly.

This used to work with target `netstandard2.0` instead.

The workaround is to set the following to false:

```xml
    <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
```

However it's rather counter-intuitive that to generate a reference assembly - but __only__ a reference assembly - you actually have to turn off producing a reference assembly.

It is likely related to this bit and explains why netstandard2.0 works but net5.0+ doesn't:
https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L233

### PR DESCRIPTION
**Pull request by marcpopMSFT**
Fix #16910

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs [hunk 1]
```diff
@@ -860,5 +860,62 @@ public void It_errors_when_publishing_single_file_without_apphost()
                 .Pass();
         }
 
+        [Fact]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_set()
```
Comment (sfoslund) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs:864]: nit: seems like these tests could easily be condensed into a theory

**Review by sfoslund — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 2]
```diff
@@ -230,7 +230,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup>
-    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))" >true</ProduceReferenceAssembly>
+    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and ('$(ProduceOnlyReferenceAssembly)' == '' or '$(ProduceOnlyReferenceAssembly)' == 'false')" >true</ProduceReferenceAssembly>
```
Comment (Nirmal4G) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:233]:
```suggestion
    <ProduceReferenceAssembly Condition="'$(ProduceReferenceAssembly)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and '$(ProduceOnlyReferenceAssembly)' != 'true'">true</ProduceReferenceAssembly>
```

Isn't this simpler?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs [hunk 3]
```diff
@@ -860,5 +860,46 @@ public void It_errors_when_publishing_single_file_without_apphost()
...
+        [InlineData("false")]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_set(string produceOnlyReferenceAssumbly)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "MainProject",
+                TargetFrameworks = "net5.0",
+                IsSdkProject = true,
+                IsExe = true
+
```
Comment (sfoslund) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs:874]: nit: empty line

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs [hunk 4]
```diff
@@ -860,5 +860,46 @@ public void It_errors_when_publishing_single_file_without_apphost()
                 .Pass();
         }
 
+        [Theory]
+        [InlineData("true")]
+        [InlineData("false")]
+        public void It_builds_the_project_successfully_with_only_reference_assembly_set(string produceOnlyReferenceAssumbly)
```
[the remark this answers is not in the document]
Comment (Nirmal4G) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs:866]: Also, it's `Assembly`!

**Review by sfoslund — APPROVED** [review]

**Comment by dotMorten** [comment]
Thank you! ❤️

**Pull request merged**